internal/mode/static/state/graph/gatewayclass.go

```diff
@@ -75,10 +76,10 @@ func buildGatewayClass(gc *v1beta1.GatewayClass) *GatewayClass {
 	}
 }
 
-func validateGatewayClass(gc *v1beta1.GatewayClass) error {
-	if gc.Spec.ParametersRef != nil {
+func validateGatewayClass(gc *v1beta1.GatewayClass, npCfg *ngfAPI.NginxProxy) error {
+	if gc.Spec.ParametersRef != nil && npCfg == nil {
```

Comment:
can we validate ParameterRef?  I think it makes sense to ensure the Kind is supported